### PR TITLE
[COST-3808] add external-id to sources integration

### PR DESCRIPTION
**Pull request by maskarb**
## Jira Ticket

[COST-3808](https://issues.redhat.com/browse/COST-3808)

## Description

This change will grab the external-id from the sources `extras` field when it is present.

### CODE REVIEW

File: koku/sources/sources_http_client.py [hunk 1]
```diff
@@ -221,20 +218,9 @@ def _get_aws_credentials(self, _):
...
-
-        auth_id = auth_data.get("id")
-        auth_internal_url = (
-            f"{self._internal_url}/{ENDPOINT_AUTHENTICATIONS}/{auth_id}?expose_encrypted_attribute[]=password"
-        )
-        auth_internal_response = self._get_network_response(auth_internal_url, "Unable to get AWS RoleARN")
-        password = auth_internal_response.get("password")
-        if password:
-            return {"role_arn": password}
+            return {"role_arn": username, "external_id": auth_data.get("extra", {}).get("external_id")}
```
Comment (samdoran) [inline koku/sources/sources_http_client.py:222]: Is it ok for `external_id` to be `None`?

Comment (maskarb) [reply koku/sources/sources_http_client.py:222]: for now, yes.

**Review by lcouzens — APPROVED** [review]
lgtm

**Review by samdoran — APPROVED** [review]

**Comment by myersCody** [comment]
@maskarb You can still merge if you want, I just didn't know if you saw eva comment about how hot fix smokes doesn't add any aws sources. So I went ahead and rekick them off for you in case you wanted to rerun them so that way you don't have to wait.

**Pull request merged**